olga/analytics/views.py

```diff
@@ -82,55 +189,54 @@ def create_instance_data(self, received_data, secret_token):  # pylint: disable=
-            students_per_country_encoded = json.dumps(self.update_students_without_country_value(
-                active_students_amount_month, students_per_country_decoded
-            ))
-
-            enthusiast_edx_installation = {
-                'latitude': float(received_data.get('latitude')),
-                'longitude': float(received_data.get('longitude')),
-                'platform_name': received_data.get('platform_name'),
-                'platform_url': received_data.get('platform_url'),
-            }
-
-            enthusiast_installation_statistics = {  # pylint: disable=invalid-name
-                'students_per_country': students_per_country_encoded
-            }
-
-            edx_installation.update(enthusiast_edx_installation)
-            installation_statistics.update(enthusiast_installation_statistics)
-
-        edx_installation_object, _ = EdxInstallation.objects.get_or_create(
-            platform_url=edx_installation['platform_url'],
-            defaults=edx_installation
-        )
+        if statistics_level == 'enthusiast':
+            self.extend_statistics_to_enthusiast_level(received_data, installation_statistics, edx_installation_object)
 
         InstallationStatistics.objects.create(edx_installation=edx_installation_object, **installation_statistics)
 
+    @staticmethod
+    def is_access_token_authorized(access_token):
+        """
+        Check if access token belongs to any EdxInstallation object.
+        """
+        try:
+            EdxInstallation.objects.get(access_token=access_token)
+            logger.info('edX installation with token %s was successfully authorized', access_token)
+            return True
+        except EdxInstallation.DoesNotExist:
+            return False
+
+    @method_decorator(installation_statistics_forms_checker)
     def post(self, request):
         """
-        Receives information from the edx-platform and processes it.
-
-        If the first time gets information from edx-platform, generates a secret token and
-        sends it back to the edx-platform for further data exchange abilities, otherwise
-        updates data in the DB with the new incoming information from the edx-platform.
+        Receive edX installation statistics and create corresponding data in database.
 
-        Returns HTTP-response with status 201, that means object (instance data) was successfully created.
+        Returns HTTP-response with status 201, that means object (installation data) was successfully created.
+        Returns HTTP-response with status 401, that means edX installation is not authorized via token.
         """
-
         received_data = request.POST
-        platform_url = received_data.get('platform_url')
-        secret_token = received_data.get('secret_token')
+        access_token = received_data.get('access_token')
+
+        if self.is_access_token_authorized(access_token):
```

Comment:
you have already "access" word twice here. So you can rename function to  "is_token_authorized"